source/MaterialXShaderGen/ShaderGenerators/Glsl/GeomColorGlsl.cpp

```diff
@@ -0,0 +1,69 @@
+#include <MaterialXShaderGen/ShaderGenerators/Glsl/GeomColorGlsl.h>
+
+namespace MaterialX
+{
+
+SgImplementationPtr GeomColorGlsl::creator()
+{
+    return std::make_shared<GeomColorGlsl>();
+}
+
+void GeomColorGlsl::createVariables(const SgNode& node, ShaderGenerator& shadergen, Shader& shader_)
+{
+    HwShader& shader = static_cast<HwShader&>(shader_);
+
+    const SgOutput* output = node.getOutput();
+    const SgInput* indexInput = node.getInput(INDEX);
+    const string index = indexInput ? indexInput->value->getValueString() : "0";
+    const string type = shadergen.getSyntax()->getTypeName(node.getOutput()->type);
+
+    if (output->type == DataType::COLOR4)
+    {
+        shader.createAppData(type, "i_color4_" + index);
+        shader.createVertexData(type, "color4_" + index);
+    }
+    else
+    {
+        shader.createAppData(type, "i_color3_" + index);
+        shader.createVertexData(type, "color3_" + index);
+    }
```

Comment:
Should we support color2 and color1 as well. Guess it would be greyscale, and greyscale + alpha? Also shouldn't this just be "i_color<index>". We don't need the 3/4 since the type gives the channel count.

Reply:
Never mind about the color1, and color2 part of the comment, I see you swizzle it out later on and we had the chat about what to send over before.